src/GafferImage/ImageStats.cpp

```diff
@@ -198,98 +222,219 @@ const DeepState *ImageStats::deepState() const
 void ImageStats::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
 {
 	ComputeNode::affects( input, outputs );
+
 	if(
 		input == flattenedInPlug()->dataWindowPlug() ||
-		input == flattenedInPlug()->channelNamesPlug() ||
 		input == flattenedInPlug()->channelDataPlug() ||
 		input == channelsPlug() ||
 		areaPlug()->isAncestorOf( input )
 	)
+	{
+		outputs.push_back( tileStatsPlug() );
+	}
+	else if(
+		input == tileStatsPlug() ||
+		input == flattenedInPlug()->dataWindowPlug() ||
+		areaPlug()->isAncestorOf( input )
+	)
+	{
+		outputs.push_back( allStatsPlug() );
+	}
+	else if(
+		input == allStatsPlug() ||
+		input == flattenedInPlug()->channelNamesPlug()
+	)
 	{
 		for( unsigned int i = 0; i < 4; ++i )
 		{
 			outputs.push_back( minPlug()->getChild(i) );
 			outputs.push_back( averagePlug()->getChild(i) );
 			outputs.push_back( maxPlug()->getChild(i) );
 		}
-		return;
 	}
 }
 
 void ImageStats::hash( const ValuePlug *output, const Context *context, IECore::MurmurHash &h ) const
 {
 	ComputeNode::hash( output, context, h);
 
-	const int colorIndex = ::colorIndex( output );
-	if( colorIndex == -1 )
+	const Plug *parent = output->parent<Plug>();
+	if( parent == minPlug() || parent == maxPlug() || parent == averagePlug() )
 	{
-		// Not a plug we know about
+		const int colorIndex = ::colorIndex( output );
+		if( colorIndex == -1 )
+		{
+			// Not a plug we know about
+			return;
+		}
```

Comment:
I know this comes from the original code, but it's unnecessarily defensive. If anything it should be an assert in `colorIndex()` because there is no way we should ever hit this. If we do the refactor of `colorIndex/channelName` I suggested above, let's clean this up too.